### DISCUSSION TITLE
Lesson 1: ParserError: Expected ',' but got 'Number'  function Store (unit 256 _favoriteNumber) public {

### DISCUSSION THREAD

**Discussion by LegalTechsis** [smartcontractkit/full-blockchain-solidity-course-py#1320] [Q&A] [answered]
Please can anyone help me identify what the error is? and how to fix it? The error is as follows; 

from solidity:
contracts/SimpleStorage1.sol:10:23: ParserError: Expected ',' but got 'Number'
 function Store (unit 256 _favoriteNumber) public {
                      ^-^

<img width="695" alt="Simplestorage error" src="https://user-images.githubusercontent.com/98588810/159280088-9603ea67-f4c3-4150-9a66-b5c933f08c3a.PNG">

but when i try to do it like the screenshot below; there are no errors; that is; close the parenthesis opened in number 5 at number 9. but after i click on deploy i can only see the button for favorite number but not the store button.

![image](https://user-images.githubusercontent.com/98588810/159283509-88923dbb-06fb-4ae0-bc68-794620ba7c25.png)

**Comment by cromewar** [comment, marked as answer]
Hello @LegalTechsis 
You have two errors:
Is `uint` no `unit`
and you have a space between it and the number, the correct form is:
`uint256` all together.

**Comment by Noveleader** [comment]
Hey @LegalTechsis change ```uint 256``` to ```uint256``` in function store
This is the reason why you getting error as after ```uint``` there is some space which means new ```function argument``` starts...Hence the error.
Hope it helps.
